packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -678,6 +689,24 @@ async function backup(projectPath: string): Promise<void> {
   }
 }
 
+// append folder path to .gitignore under the project root.
+async function addPathToGitignore(projectPath: string, ignoredPath: string) {
+  const relativePath = path.relative(projectPath, ignoredPath).replace(/\\/g, "/");
+  await addItemToGitignore(projectPath, relativePath);
+}
+
+// append item to .gitignore under the project root.
+async function addItemToGitignore(projectPath: string, item: string) {
+  const gitignorePath = path.join(projectPath, gitignoreFileName);
```

Comment:
It is better to add try catch for losing write permission. 

```suggestion
try{
     const gitignorePath = path.join(projectPath, gitignoreFileName);
     await fs.ensureFile(gitignorePath);
  
    const gitignoreContent = await fs.readFile(gitignorePath, "UTF-8");
    if (gitignoreContent.indexOf(item) === -1) {
      const appendedContent = os.EOL + item;
      await fs.appendFile(gitignorePath, appendedContent);
    }
  }catch(err) {
   // do nothing
  }
```

Reply:
Shall we report this error instead of hiding it, otherwise .backup might be checked-in by mistake?

Reply:
Offline discussed, we're going to catch the error and add user warning to let user do it manually without blocking the migration.